annotations/views/project_views.py

```diff
@@ -60,6 +60,19 @@ def add_text(self, request, pk=None):
         serializer = ProjectSerializer(project)
         return Response(serializer.data)
 
+    @action(detail=True, methods=['DELETE'], url_name='deletetext')
```

Comment:
There is no need to create a custom action here. Viewsets give us a destroy action by default. You can see a list of default actions here https://www.django-rest-framework.org/api-guide/viewsets/#viewset-actions. You can override the default destroy method to implement what you need. This way we do not have unneeded routes and we are leveraging the full usefulness of Viewsets.